lib/query-interface.js

```diff
@@ -880,6 +880,25 @@ module.exports = (function() {
     return this.sequelize.query(sql, options);
   };
 
+  QueryInterface.prototype.deferConstraints = function (transaction, options) {
+    if (!(transaction instanceof Transaction)) {
+      throw new Error('Unable to start a transaction without transaction object!');
+    }
```

Comment:
I'd say this if is not necessary. The internal call can't fail and if someone wants to use this directly that person should know to insert the transaction :)